### PR TITLE
[SofaCarving] plugin enhancement

### PR DESCRIPTION
**Pull request by epernod**
- Add: script event listener
- Add: scene coupling Geomagic and SofaCarving 
- Update carving detection collision to use the existing pipeline instead of relaunching a narrowphase intersection computation.

- Add method to remove topological element from PointModel collision.
- Fix sofaCarving tests

______________________________________________________

This PR: 
- [x] builds with SUCCESS for all platforms on the CI.
- [x] does not generate new warnings.
- [x] does not generate new unit test failures.
- [x] does not generate new scene test failures.
- [x] does not break API compatibility.
- [x] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by epernod** [comment]
[ci-build][with-scene-tests]

File: applications/plugins/SofaCarving/CarvingManager.cpp [hunk 1]
```diff
@@ -205,13 +194,13 @@ void CarvingManager::doCarve()
...
         }
     }
-    
-
-    m_detectionNP->setInstance(NULL);
 }
 
 void CarvingManager::handleEvent(sofa::core::objectmodel::Event* event)
 {
+    if (m_carvingReady == false)
```
Comment (damienmarchal) [inline applications/plugins/SofaCarving/CarvingManager.cpp]: why not as in line 140 ?

File: applications/plugins/SofaCarving/CarvingManager.cpp [hunk 2]
```diff
@@ -248,12 +237,21 @@ void CarvingManager::handleEvent(sofa::core::objectmodel::Event* event)
         if (ev->getButtonState()==1) d_active.setValue(true);
```
Comment (damienmarchal) [inline applications/plugins/SofaCarving/CarvingManager.cpp:237]: d_active.setValue(ev->getButtonState()); ?

unless getButtonState is return other things that 1 or 0...I'm probably not well awake as I don't see this function in my code base.

File: applications/plugins/SofaCarving/CarvingManager.cpp [hunk 3]
```diff
@@ -248,12 +237,21 @@ void CarvingManager::handleEvent(sofa::core::objectmodel::Event* event)
...
         else if (ev->getButtonState()==0) d_active.setValue(false);
     }
-
-    else if (/* simulation::AnimateEndEvent* ev = */ dynamic_cast<simulation::AnimateEndEvent*>(event))
+    else if (sofa::core::objectmodel::ScriptEvent *ev = dynamic_cast<sofa::core::objectmodel::ScriptEvent *>(event))
+    {
+        const std::string& eventS = ev->getEventName();
+        if (eventS.find(d_buttonName.getValue()) != std::string::npos && eventS.find("pressed") != std::string::npos)
```
Comment (damienmarchal) [inline applications/plugins/SofaCarving/CarvingManager.cpp]: Here again.

File: applications/plugins/SofaCarving/CarvingManager.h [hunk 4]
```diff
@@ -99,6 +99,8 @@ class SOFA_SOFACARVING_API CarvingManager : public core::behavior::BaseControlle
     Data < bool > d_mouseEvent;
     ///< Activate carving with omni button
     Data < bool > d_omniEvent;
+    ///< Activate carving with string Event
+    Data < std::string > d_buttonName;
```
Comment (damienmarchal) [inline applications/plugins/SofaCarving/CarvingManager.h]: eventName ?

**Pull request merged**